kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -431,16 +419,35 @@ public List<EventSource<?, VirtualKafkaCluster>> prepareEventSources(EventSource
                 Service.class,
                 VirtualKafkaCluster.class)
                 .withName(KUBERNETES_SERVICES_EVENT_SOURCE_NAME)
-                .withPrimaryToSecondaryMapper((VirtualKafkaCluster cluster) -> cluster.getSpec().getIngresses()
-                        .stream()
-                        .map(Ingresses::getIngressRef)
-                        .flatMap(ir -> ResourcesUtil
-                                .localRefAsResourceId(cluster, new AnyLocalRefBuilder().withName(bootstrapServiceName(cluster, ir.getName())).build()).stream())
-                        .collect(Collectors.toSet()))
-                .withSecondaryToPrimaryMapper(kubenetesService -> Optional.of(kubenetesService)
-                        .flatMap(service -> extractOwnerRefFromKubernetesService(service, VIRTUAL_KAFKA_CLUSTER_KIND))
-                        .map(ownerRef -> new ResourceID(ownerRef.getName(), kubenetesService.getMetadata().getNamespace()))
-                        .map(Set::of).orElse(Set.of()))
+                .withPrimaryToSecondaryMapper((VirtualKafkaCluster cluster) -> {
+                    Stream<ResourceID> clusterIpServices = cluster.getSpec().getIngresses()
+                            .stream()
+                            .map(Ingresses::getIngressRef)
+                            .flatMap(ir -> ResourcesUtil
+                                    .localRefAsResourceId(cluster, new AnyLocalRefBuilder().withName(bootstrapServiceName(cluster, ir.getName())).build()).stream());
+                    Stream<ResourceID> loadbalancerService = ResourcesUtil.localRefAsResourceId(cluster,
+                            new AnyLocalRefBuilder().withName(cluster.getSpec().getProxyRef().getName() + "-sni").build()).stream();
```

Comment:
Are we adding a suffix which we control to every `Service`? Or otherwise how can we guarantee there will be no collision if the user causes some other Service to be created which happens to end with `-sni`?

Reply:
Having `-bootstrap` seems like a good idea. Are there any changes which might happen in the future which could also collide within the space of `Service` names?

Reply:
assuming that's what you're talking about, collisions between services the Operator controls.

Reply:
Created #2283 we can continue discussion there about any future Services we can think of.